src/escpos/escpos.py

```diff
@@ -422,6 +453,128 @@ def barcode(
+
+        :param height: barcode module height (in printer dots), has to be between 1 and 255.
+            *default*: 64
+        :type height: int
+
+        :param width: barcode module width (in printer dots), has to be between 2 and 6.
+            *default*: 3
+        :type width: int
+
+        :param pos: text position (ABOVE, BELOW, BOTH, OFF) relative to the barcode
+            (ignored in software renderer).
+            *default*: BELOW
+
+        :param font: select font A or B (ignored in software renderer).
+            *default*: A
+
+        :param align_ct: If *True*, center the barcode.
+            *default*: True
+        :type align_ct: bool
+
+        :param function_type: ESCPOS function type A or B. None to guess it from profile
+            (ignored in software renderer).
+            *default*: None
+
+        :param check: If *True*, checks that the code meets the requirements of the barcode type.
+            *default*: True
+        :type check: bool
+
+        :param force_software: If *True*, force the use of software barcode renderer from profile.
+            If *"graphics", "bitImageColumn" or "bitImageRaster"*, force the use of specific renderer.
+        :type force_software: bool | str
+
+        :raises: :py:exc:`~escpos.exceptions.BarcodeCodeError`,
+                 :py:exc:`~escpos.exceptions.BarcodeTypeError`
+
+        .. tip::
+            Get all supported formats at:
+            - Hardware: `~escpos.constants.BARCODE_FORMATS`
+            - Software: `https://python-barcode.readthedocs.io/en/stable/supported-formats.html`
+        """
+        hw_modes = ["barcodeA", "barcodeB"]
+        sw_modes = ["graphics", "bitImageColumn", "bitImageRaster"]
+        capable = {
+            "hw": [mode for mode in hw_modes if self.profile.supports(mode)] or None,
+            "sw": [mode for mode in sw_modes if self.profile.supports(mode)] or None,
+        }
+        if (not capable["hw"] and not capable["sw"]) or (
+            not capable["sw"] and force_software
+        ):
+            raise BarcodeTypeError(
+                f"""Profile {
+                    self.profile.profile_data['name']
+                } - hw barcode: {capable['hw']}, sw barcode: {capable['sw']}"""
+            )
+
+        bc_alnum = "".join([char for char in bc.upper() if char.isalnum()])
+        capable_bc = {
+            "hw": HW_BARCODE_NAMES.get(bc_alnum),
+            "sw": SW_BARCODE_NAMES.get(bc_alnum),
+        }
+        if not any([*capable_bc.values()]):
+            raise BarcodeTypeError(f"Not supported or wrong barcode name {bc}.")
+
+        if force_software or not capable["hw"] or not capable_bc["hw"]:
+            # Select the best possible capable render mode
+            impl = capable["sw"][0]
+            if force_software in capable["sw"]:
+                # Force to a specific mode
+                impl = force_software
+            print(f"Using {impl} software barcode renderer")
```

Comment:
I would prefer for messages like this a logging facility, but as of now we do not have it in python-escpos and therefore I would just keep it as it is and rework all print statements at a later point.